src/oid4vp/request_signer.rs

```diff
@@ -0,0 +1,19 @@
+#[derive(Debug, thiserror::Error, uniffi::Error)]
+pub enum RequestSignerError {
+    #[error("Unsupported algorithm")]
+    UnsupportedAlgorithm,
+    #[error("Failed to sign the request")]
+    SigningError,
+}
+
+#[uniffi::export(with_foreign)]
+pub trait RequestSignerInterface: Send + Sync + std::fmt::Debug {
+    /// Return the algorithm used to sign the request
+    fn alg(&self) -> Result<String, RequestSignerError>;
+
+    /// Return the JWK public key
+    fn jwk(&self) -> Result<String, RequestSignerError>;
+
+    /// Sign the request
+    fn try_sign(&self, payload: Vec<u8>) -> Result<Vec<u8>, RequestSignerError>;
+}
```

Comment:
TODO: Add comments for this interface.

Reply:
I've removed this interface, we may need to re-add something like this for the verifier side, but we can add in another PR. It was not needed for the Holder.